src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureCount = random.nextInt(FIGURE_COUNT);
+        FigureList[] values = FigureList.values();
+        FigureList randomFigureType = values[figureCount];
+        ColorSupplier colorSupplier = new ColorSupplier();
+        String randomColor = colorSupplier.getRandomColor();
+
+        switch (randomFigureType) {
+            case CIRCLE:
+                double radius = random.nextInt(100);
+                return new Circle(radius, randomColor);
```

Comment:
lets
```suggestion
            case CIRCLE:
                return getRandomCircle();
```
To make your code easier to read - lets add private methods to get each random figure. 
Also remember to avoid magic numbers - use constants